### PR TITLE
bitlbee-mastodon: init at 1.4.2

### PR DESCRIPTION
**Pull request by jpotier**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] (NA) Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] (NA) Tested execution of all binary files (usually in `./result/bin/`)
- [ ] (NA) Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jb55** [comment]
another dejavu moment when I'm about to package something but I check here first. nixpkgs is becoming hive-mind like. Testing now.

File: pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  preConfigure = ''
+    export BITLBEE_PLUGINDIR=$out/lib/bitlbee
+    export BITLBEE_DATADIR=$out/share/bitlbee
+    ./autogen.sh
+  '';
+
+  meta = {
+    description = "Bitlbee plugin for Mastodon";
+
```
Comment (jb55) [inline pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix]: nit: empty line could be removed

```suggestion
```

**Review by jb55 — CHANGES_REQUESTED** [review]

**Review by jb55 — APPROVED** [review]

**Comment by jpotier** [comment]
Should I pick reviewers/approvers manually in order to move this forward?

**Comment by jb55** [comment]
fwiw I've been using this for the past 17 days without any issues.

**Comment by chreekat** [comment]
This is implemented exactly the same as the other bitlbee plugins, lgtm

**Review by Lassulus — APPROVED** [review]

File: pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ fetchgit, stdenv, bitlbee, autoreconfHook, pkgconfig, glib }:
+
+with stdenv.lib;
+stdenv.mkDerivation rec {
+  name = "bitlbee-mastodon-${version}";
```
Comment (aanderse) [inline pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix]: Please replace with `pname = "bitlbee-mastodon";` as `pname = "${name}-${version}";` is already implied.

File: pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  preConfigure = ''
+    export BITLBEE_PLUGINDIR=$out/lib/bitlbee
+    export BITLBEE_DATADIR=$out/share/bitlbee
+    ./autogen.sh
+  '';
+
+  meta = {
+    description = "Bitlbee plugin for Mastodon";
+    homepage = https://alexschroeder.ch/cgit/bitlbee-mastodon/about;
```
Comment (aanderse) [inline pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix]: Please quote.

**Review by aanderse — CHANGES_REQUESTED** [review]
2 very minor nitpicks, but yeah this should be merged. :+1:

File: pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix (hunk 2 above)
Comment (jpotier) [reply pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix]: I didn't know about that, done.

File: pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix (hunk 3 above)
Comment (jpotier) [reply pkgs/applications/networking/instant-messengers/bitlbee-mastodon/default.nix]: Done

**Comment by aanderse** [comment]
@GrahamcOfBorg build bitlbee-mastodon

**Review by aanderse — APPROVED** [review]

**Pull request merged**